internal/controller/status.go

```diff
@@ -2,11 +2,93 @@ package controller
 
 import (
 	"context"
+	"fmt"
+	"slices"
 
 	rsctv1alpha1 "github.com/ocp-power-automation/rsct-operator/api/v1alpha1"
 	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type PodStatus string
+
+const (
+	PENDING           PodStatus = "PENDING"
+	RUNNING           PodStatus = "RUNNING"
+	FAILED            PodStatus = "RUNNING"
+	PARTIALLY_RUNNING PodStatus = "PARTIALLY_RUNNING"
+	UNKNOWN           PodStatus = "UNKNOWN"
+)
+
+// matchPodsStatus checks if status of all the pods in slice are same or not
+func matchPodsStatus(podsStatus []string, status string) bool {
+	for _, ps := range podsStatus {
+		if ps == status {
+			continue
+		}
+		return false
+	}
+	return true
+}
+
+// evalOperatorStatus determines operator status based on the pods status
+func evalOperatorStatus(podList *corev1.PodList) string {
+	var effectiveStatus string
+	var podsStatus []string
```

Comment:
can this be `[]PodStatus` and avoid string casting below

Reply:
makes sense. will do